File: basic/1071. 小赌怡情(15).cpp
#include<iostream>
#include<cmath>
#include<cstdio>
#include<algorithm>
using namespace std;
const double eps = 1e-8;
const int maxn = 110;

int main() {
	int T, k, n1, b, t, n2;
	cin >> T >> k;
	while (k--) {
		cin >> n1 >> b >> t >> n2;
		if(T<t)printf("Not enough tokens.");
		else {
			if ((n2 > n1&&b) || (n2 < n1 && !b)) {
				printf("Win %d!", t); T += t;
			}
			else if (T >= t) {
				printf("Lose %d.", t); T -= t;
			}
		}
		printf("  Total = %d.\n", T);
		if (T <= 0) { printf("Game Over."); return 0; }
	}
	return 0;
}

File: basic/java/1071. 小赌怡情(15).java
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner in=new Scanner(System.in);
		int t=in.nextInt(),k=in.nextInt();
		for(int i=0;i<k;++i) {
			int n1=in.nextInt(),b=in.nextInt(),tt=in.nextInt(),n2=in.nextInt();
			if(t<tt)System.out.print("Not enough tokens.");
			else {
				if((n2>n1&&b!=0)||(n2<n1&&b==0)) {
					System.out.print("Win "+tt+"!");
					t+=tt;
				}else if(t>=tt) {
					System.out.print("Lose "+tt+".");
					t-=tt;
				}
			}
			System.out.println("  Total = "+t+".");
			if(t<=0) {
				System.out.print("Game Over.");
				break;
			}
		}
		in.close();
	}
	
}
